File: WindPowerSystemV5.Server/Data/MongoDbModels/News.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace WindPowerSystemV5.Server.Data.MongoDbModels;

public class News
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = null!;

    public string Title { get; set; } = null!;

    public string Author { get; set; } = null!;

    public List<string> Chapters { get; set; } = [];

    public DateTime CreatedDt { get; set; }
    
    public DateTime? UpdatedDt { get; set; }

    public List<NewsComment> Comments { get; set; } = [];

    public int Likes { get; set; }
}

File: windpowersystemv5.client/src/app/news/news.ts
import { NewsComment } from './news-comment';

export interface News {
  id: string;
  title: string;
  author: string;
  chapters: string[];
  createdDt: string; // ISO date string
  updatedDt?: string; 
  comments: NewsComment[];
  likes: number;
}
